### DISCUSSION TITLE
'components directory in the app directory'  vs 'components directory in the src directory'

### DISCUSSION THREAD

**Discussion by kwonmory** [vercel/next.js#50490] [App Router] [answered]
Hello! 

I had a question during development.

To use as a server component. I'm creating a 'components' directory in 'app' directory and placing the components there.

However, I wonder if the component will be used as a server component even if I use it outside the app directory.

I am curious about the difference between the two.
```
src
 - app
   ㄴ- components
```

```
src
 - app
 - components
```

**Comment by devjiwonchoi** [comment, marked as answer]
Based on this [docs](https://nextjs.org/docs/app/building-your-application/routing/colocation#project-organization-strategies), it does not matter whether you put the `components` folder inside or outside the `app` directory.
>There is no "right" or "wrong" way when it comes to organizing your own files and folders in Next.js a project.

For example in this case:
```
src
 - app
 - components
```
 Since you call your components from the `components` folder to the `app` folder, 
the components will be a Server Component by default unless you set `'use client';` on the top of the component.
> To make the transition to Server Components easier, all components inside the [App Router](https://nextjs.org/docs/app/building-your-application/routing#the-app-directory) are Server Components by default, including [special files](https://nextjs.org/docs/app/building-your-application/routing#file-conventions) and [colocated components](https://nextjs.org/docs/app/building-your-application/routing#colocation).

To check if your component is a Server Component or Client Component, 
try `console.log()` to see if the log is displayed on the Browser's console or just on your IDE's console.

**Reply by kwonmory** [reply]
First of all, thank you for your reply.

I got the answer I wanted. Thank you!! :)

**Comment by raferdev** [comment]
I think the difference is more conceptual, the folder inside app directory says the component are used to build the top level layout of the app, if i have more paths inside the app, with pages, i create a new component folder to store this components that are only used to build each page, on the other hand if i have a component that are shared between two or more pages, i create a component folder direcly in src directory, that says is a global component. This is my way to build but not the only. In the end i have something like this:

src:
 - components (global)
 - app:
     - components (top level app layout components)
     - [page1]
         - components (page 1 components)
         - page.tsx
         - ...
     - [page2]
         - components (page 2 components)
         - page.tsx

**Reply by devjiwonchoi** [reply]
This is one of the Project organization strategies called [Colocation](https://nextjs.org/docs/app/building-your-application/routing#colocation).

You can find a similar example at [here](https://nextjs.org/docs/app/building-your-application/routing/colocation#project-organization-strategies).

**Reply by raferdev** [reply]
That is it

**Reply by kwonmory** [reply]
I like it because it seems like a foldering structure considering locality.

I was doing it similar to @imraferdev.

But I didn't think of it as global components.

Thank you for the good tips. : )

I'll develop it with reference.

**Comment by kwonmory** [comment]
Thank you to everyone who responded.
